### PR TITLE
Update eslint (major) & Convert eslint configuration file

### PR DESCRIPTION
**Pull request by semnil**
## Problem to solve

- Since eslint 9, flat config is the default and lint fails
```
Run pnpm lint

> monorepo-deploy-actions@ lint /home/runner/work/monorepo-deploy-actions/monorepo-deploy-actions
> eslint --fix **/*.ts


Oops! Something went wrong! :(

ESLint: [9](https://github.com/quipper/monorepo-deploy-actions/actions/runs/8586678434/job/23529426235#step:7:10).0.0

Error: Could not find config file.
...
```
https://github.com/quipper/monorepo-deploy-actions/actions/runs/8586678434/job/23529426235
- Convert the configuration file to flat config
  - https://eslint.org/docs/latest/use/configure/migration-guide

### CODE REVIEW

File: eslint.config.mjs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+// @ts-check
+
+import eslint from '@eslint/js';
+import jest from 'eslint-plugin-jest';
+import tseslint from 'typescript-eslint';
+import typescriptEslintParser from "@typescript-eslint/parser";
+
+export default tseslint.config(
+  eslint.configs.recommended,
```
Comment (semnil) [inline eslint.config.mjs:9]: https://github.com/quipper/monorepo-deploy-actions/blob/e394a26a8937e2761d099e0107efe33dd60d1077/.eslintrc.json#L3

File: eslint.config.mjs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  ...tseslint.configs.recommended,
+  ...tseslint.configs.recommendedTypeChecked,
```
Comment (semnil) [inline eslint.config.mjs:11]: https://github.com/quipper/monorepo-deploy-actions/blob/e394a26a8937e2761d099e0107efe33dd60d1077/.eslintrc.json#L5-L6

File: eslint.config.mjs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  {
+    files: ['tests/**'],
+    ...jest.configs['flat/recommended'],
+    rules: {
+      ...jest.configs['flat/recommended'].rules,
```
Comment (semnil) [inline eslint.config.mjs:16]: https://github.com/quipper/monorepo-deploy-actions/blob/e394a26a8937e2761d099e0107efe33dd60d1077/.eslintrc.json#L4

File: eslint.config.mjs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    },
+  },
+  {
+    languageOptions: {
+      parser: typescriptEslintParser,
+      parserOptions: {
+        project: true,
+      },
```
Comment (semnil) [inline eslint.config.mjs:24]: https://github.com/quipper/monorepo-deploy-actions/blob/e394a26a8937e2761d099e0107efe33dd60d1077/.eslintrc.json#L9-L12

File: .eslintrc.json [hunk 5]
```diff
@@ -1,14 +0,0 @@
...
-    "plugin:jest/recommended",
-    "plugin:@typescript-eslint/recommended",
-    "plugin:@typescript-eslint/recommended-requiring-type-checking"
-  ],
-  "plugins": ["@typescript-eslint", "jest"],
-  "parser": "@typescript-eslint/parser",
-  "parserOptions": {
-    "project": true
-  },
-  "root": true
```
Comment (semnil) [inline .eslintrc.json:13]: > The root option no longer exists. (Flat config files act as if root: true is set.)
https://eslint.org/docs/latest/use/configure/migration-guide#additional-changes

File: package.json [hunk 6]
```diff
@@ -3,18 +3,19 @@
   "name": "monorepo-deploy-actions",
   "license": "MIT",
   "devDependencies": {
+    "@eslint/js": "^9.0.0",
```
Comment (semnil) [inline package.json:6]: Fix Version
```suggestion
    "@eslint/js": "9.0.0",
```

File: package.json [hunk 7]
```diff
@@ -3,18 +3,19 @@
...
-    "eslint": "8.57.0",
-    "eslint-plugin-jest": "27.9.0",
+    "eslint": "9.0.0",
+    "eslint-plugin-jest": "28.2.0",
     "jest": "29.7.0",
     "prettier": "3.2.5",
     "ts-jest": "29.1.2",
-    "typescript": "5.4.4"
+    "typescript": "5.4.4",
+    "typescript-eslint": "^7.5.0"
```
Comment (semnil) [inline package.json:18]: Fix Version
```suggestion
    "typescript-eslint": "7.5.0"
```

**Review by int128 — APPROVED** [review]

**Pull request merged**